package DB;

import java.sql.SQLException;

public class DBManagement {

	SqlTest sql = new SqlTest();

	public void show() {
		try {
			String cmd = "select * from product";

			sql.st = sql.con.createStatement();
			sql.rs = sql.st.executeQuery(cmd);

			while (sql.rs.next()) {
				String name = sql.rs.getString("name");
				int price = sql.rs.getInt("price");
				System.out.printf("name : %s, price : %d\n", name, price);
			}
		} catch (SQLException e) {
			e.printStackTrace();
		}
	}

	public void add(String name, int price) {
		try {
			String cmd = "INSERT INTO PRODUCT (name, price)" + "VALUES ('" + name + "', '" + price + "')";

			sql.st = sql.con.createStatement();
			int result = sql.st.executeUpdate(cmd);

			System.out.println(result + "개가 성공적으로 추가되었습니다.");
		} catch (SQLException e) {
			System.out.println(name + "와(과) 같은 상품이 있습니다.");
			// e.printStackTrace();
			return;
		}
	}
	
	public void delete(String name) {
		try {
			String cmd = "DELETE FROM PRODUCT WHERE NAME = '"+ name + "'";
			
			sql.st = sql.con.createStatement();
			int result = sql.st.executeUpdate(cmd);
			
			System.out.println(result + "개가 성공적으로 삭제되었습니다.");
		} catch (SQLException e) {
			e.printStackTrace();
		}
	}

}
